fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/item/v1/FabricItemSettings.java

```diff
@@ -53,6 +55,55 @@ public FabricItemSettings customDamage(CustomDamageHandler handler) {
 		return this;
 	}
 
+	/**
+	 * Sets the stack-aware recipe remainder provider of the item.
+	 * @see RecipeRemainderProvider
+	 */
+	public FabricItemSettings recipeRemainder(RecipeRemainderProvider provider) {
+		FabricItemInternals.computeExtraData(this).recipeRemainderProvider(provider);
+		return this;
+	}
+
+	/**
+	 * Sets the stack-aware recipe remainder to damage the item by 1 every time it is used in crafting.
+	 */
+	public FabricItemSettings damageIfUsedInCrafting() {
+		return this.damageIfUsedInCrafting(1);
+	}
+
+	/**
+	 * Sets the stack-aware recipe remainder to return the item itself.
+	 */
+	public FabricItemSettings returnSelfInCrafting() {
+		return this.damageIfUsedInCrafting(0);
+	}
+
+	/**
+	 * Sets the stack-aware recipe remainder to damage the item by a certain amount every time it is used in crafting.
+	 *
+	 * @param by the amount
+	 */
+	public FabricItemSettings damageIfUsedInCrafting(int by) {
+		return this.recipeRemainder((original, inventory, type, world, pos) -> {
+			if (by == 0) {
+				return original;
+			}
+
+			if (!original.isDamageable()) {
+				return ItemStack.EMPTY;
+			}
+
+			ItemStack copy = original.copy();
+			copy.damage(by, ThreadLocalRandom.current(), null);
```

Comment:
Just make sure this doesn't cause game behavior changes.

Reply:
I assume that if no one calls this method, it won't take any affect, no?